src/OrleansRuntime/GrainTypeManager/GrainTypeData.cs

```diff
@@ -151,5 +154,11 @@ internal static PlacementStrategy GetPlacementStrategy(Type grainClass, Placemen
 
             return predicate;
         }
+
+        private ParameterInfo[] GetConstructorParameters()
+        {
+            var constructor = this.Type.GetConstructors(BindingFlags.Public | BindingFlags.Instance).FirstOrDefault();
+            return constructor?.GetParameters() ?? EmptyParameters;
+        }
```

Comment:
IMO it's fine to push the requirement of consistent constructor selection onto `IGrainActivator` implementations. ctor injection is my preference, even if we also support property injection later. If we do support ctor injection of facets, I don't see how else we can do it.

Does anyone have suggestions?